pilot/pkg/bootstrap/server.go

```diff
@@ -465,17 +471,25 @@ func (s *Server) Start(stop <-chan struct{}) error {
 	}
 
 	// At this point we are ready - start Http Listener so that it can respond to readiness events.
+	httpListener, err := net.Listen("tcp", s.httpServer.Addr)
+	if err != nil {
+		return err
+	}
 	go func() {
-		log.Infof("starting Http service at %s", s.HTTPListener.Addr())
-		if err := s.httpServer.Serve(s.HTTPListener); isUnexpectedListenerError(err) {
+		log.Infof("starting HTTP service at %s", httpListener.Addr())
+		if err := s.httpServer.Serve(httpListener); isUnexpectedListenerError(err) {
 			log.Errorf("error serving http server: %v", err)
 		}
 	}()
 
 	if s.httpsServer != nil {
+		httpsListener, err := net.Listen("tcp", s.httpsServer.Addr)
+		if err != nil {
+			return err
+		}
 		go func() {
-			log.Infof("starting webhook service at %s", s.httpsServer.Addr)
-			if err := s.httpsServer.ListenAndServeTLS("", ""); isUnexpectedListenerError(err) {
+			log.Infof("starting webhook service at %s", httpsListener.Addr())
+			if err := s.httpsServer.ServeTLS(httpsListener, "", ""); isUnexpectedListenerError(err) {
```

Comment:
Serve in s separate go routine does not gurantee Listen and Serve simutaneously.

Reply:
. I think it's ok to have a tiny delay compared to today where it may be indefinitely? Not sure there is much else we can do, I think there is always some tiny delay